website/mailing_list/utils.py

```diff
@@ -0,0 +1,422 @@
+                        remove_user_from_list(node_id, u_id)
+
+        else:
+            create_list(node_id)
+    else:
+        if 'list' in info.keys():
+            delete_list(node_id)
+
+
+###############################################################################
+# Celery Queued tasks
+###############################################################################
+
+@queued_task
+@app.task
+def celery_create_list(*args, **kwargs):
+    create_list(*args, **kwargs)
+
+@queued_task
+@app.task
+def celery_delete_list(*args, **kwargs):
+    delete_list(*args, **kwargs)
+
+@queued_task
+@app.task
+def celery_update_title(*args, **kwargs):
+    update_title(*args, **kwargs)
+
+@queued_task
+@app.task
+def celery_update_single_user_in_list(*args, **kwargs):
+    update_single_user_in_list(*args, **kwargs)
+
+@queued_task
+@app.task
+def celery_remove_user_from_list(*args, **kwargs):
+    remove_user_from_list(*args, **kwargs)
+
+@queued_task
+@app.task
+def celery_update_multiple_users_in_list(*args, **kwargs):
+    update_multiple_users_in_list(*args, **kwargs)
+
+@queued_task
+@app.task
+def celery_full_update(*args, **kwargs):
+    full_update(*args, **kwargs)
+
+
+###############################################################################
+# Signalled Functions
+###############################################################################
+
+@contributor_added.connect
+def subscribe_contributor_to_mailing_list(node, contributor, auth=None):
+    if node.mailing_enabled and contributor.is_active:
+        subscription = node.get_or_create_mailing_list_subscription()
+        subscription.add_user_to_subscription(contributor, 'email_transactional', save=True)
+        celery_update_single_user_in_list(node._id, contributor._id)
+        node.mailing_updated = True
+        node.save()
+
+@contributor_removed.connect
+def unsubscribe_contributor_from_mailing_list(node, user, auth=None):
+    if node.mailing_enabled:
+        subscription = node.get_or_create_mailing_list_subscription()
+    else:
+        from website.models import NotificationSubscription  # avoid circular import
+        subscription = NotificationSubscription.load(to_subscription_key(node._id, 'mailing_list_events'))
+    if subscription:
+        subscription.remove_user_from_subscription(user)
+        node.mailing_updated = True
+        node.save()
+    celery_remove_user_from_list(node._id, user._id)
+
+
+@user_confirmed.connect
+def resubscribe_on_confirm(user):
+    for node in user.contributed:
+        subscribe_contributor_to_mailing_list(node, user)
+
+@node_deleted.connect
+def remove_list_for_deleted_node(node):
+    node.mailing_updated = True
+    node.save()
+    celery_delete_list(node._id)
+
+
+###############################################################################
+# Mailing List Helper Functions
+###############################################################################
+
+def jsonify_users_list(users, unsubs=[]):
```

Comment:
Avoid mutable default argument values in function definitions- this can lead to some truly *annoying* bugs.

Prefer `unsubs=()` instead.